vnext/Microsoft.ReactNative/IReactPackageBuilder.idl

```diff
@@ -26,18 +26,15 @@ namespace Microsoft.ReactNative
     DOC_STRING("Adds a custom native module. See @ReactModuleProvider.")
     void AddModule(String moduleName, ReactModuleProvider moduleProvider);
 
+    DOC_STRING("Adds a custom native module. See @ReactModuleProvider. This will register the"
+    "module as a TurboModule unless the application is running using @ReactInstanceSettings.UseWebDebugger,"
+    "in which case it will revert to a legacy NativeModule.\n"
+    "NOTE: TurboModules using JSI directly will not run correctly while using @ReactInstanceSettings.UseWebDebugger")
+    void AddTurboModule(String moduleName, ReactModuleProvider moduleProvider);
```

Comment:
Platforms can choose to offer a strong contract to enable independent development of the components involved. This is something my team wants to avoid tearing between our RN code and its native platform that are developed in different repositories.

midl supports `apicontract` as a way to version a design without breaking the existing use, that would look like this.

```cpp
[apicontract(ReactNativeContract, 2.0]
{
    void AddTurboModule(...);
}
```
Doing this would enable clients to detect the presence of this feature and use it or to use the experimental version if that is all that is available. under the covers the new contract would generate a new interface that can be detected at runtime using QueryInterface()

Reply:
Did you mean [contract]? or the apicontract type keyword?
the docs for [contract] says not to use it for your own apis as it only makes sense for Windows:
https://learn.microsoft.com/en-us/uwp/midl-3/predefined-attributes#the-contract-attribute
as for apicontract, that's not a publicly documented keyword AFAICT so I don't think we can use it here.